pkg/crawlers/azure/azure.go

```diff
@@ -278,12 +277,20 @@ func (c *azureNetworkCrawler) fetchAll() ([][]byte, error) {
 	// and the page then renders generated URLs to json files.
 	jsonURLs := make([]string, 0, len(c.urls))
 	for _, url := range c.urls {
-		jsonURL, err := c.redirectToJSONURL(url)
-		if err != nil {
-			return nil, errors.Wrapf(err, "failed to crawl Azure with URL: %s. Error: %v. JSON URL: %s", url, err, jsonURL)
-		}
-		if jsonURL == "" {
-			return nil, errors.Errorf("failed to crawl Azure with URL: %s, empty JSON URL returned. This could indicate Azure's services are unavailable", url)
+		var jsonURL string
+		retryErr := utils.WithDefaultRetry(func() error {
+			var err error
+			jsonURL, err = c.redirectToJSONURL(url)
+			if err != nil {
+				return errors.Wrapf(err, "failed to crawl Azure with URL: %s. Error: %v. JSON URL: %s", url, err, jsonURL)
+			}
+			if jsonURL == "" {
+				return errors.Errorf("failed to crawl Azure with URL: %s, empty JSON URL returned. This could indicate Azure's services are unavailable", url)
+			}
+			return nil
+		})
+		if retryErr != nil {
+			return nil, retryErr
 		}
 		log.Printf("Received Azure network JSON URL: %s", jsonURL)
```

Comment:
I know, this is not part of this PR, but in case there is another push planned, let's improve this message.

 ```suggestion
		log.Printf("Success obtaining Azure network JSON URL %q from %q", jsonURL, url)
```
(sorry, not sure how those 3 spaces got there into the diff - I somehow cannot remove it from the level of review comment)

Reply:
Done